src/repositories/request-repository.ts

```diff
@@ -35,164 +31,215 @@ const countRetryMetrics = (requests: Request[], anchoringDeadline: Date): void =
    */
-  public async createOrUpdate(request: Request): Promise<Request> {
-    const insertResults = await this.connection.getRepository(Request).upsert(request, ['cid'])
+  public async createOrUpdate(request: Request, options: Options = {}): Promise<Request> {
+    const { connection = this.connection } = options
+    const [{ id }] = await connection
+      .table(TABLE_NAME)
+      .insert(request, ['id'])
+      .onConflict('cid')
+      .merge()
+
+    const created = await connection.table(TABLE_NAME).first().where({ id })
+
+    logEvent.db({
+      type: 'request',
+      action: 'upsert',
+      ...request,
+      createdAt: created.createdAt.getTime(),
+      updatedAt: created.updatedAt.getTime(),
+    })
 
-    return this.connection.getRepository(Request).findOne(insertResults.identifiers[0].id)
+    return created
   }
 
   /**
-   * Creates client requests
-   * @param requests - Requests
+   * For test use. Creates an array of requests.
+   * @param requests array of requests
+   * @param options
+   * @returns
    */
-  public async createRequests(requests: Array<Request>): Promise<InsertResult> {
-    return this.connection
-      .getRepository(Request)
-      .createQueryBuilder()
-      .insert()
-      .into(Request)
-      .values(requests)
-      .execute()
+  public async createRequests(requests: Array<Request>, options: Options = {}): Promise<void> {
+    const { connection = this.connection } = options
+
+    await connection.table(TABLE_NAME).insert(requests)
   }
 
   /**
    * Create/updates client requests
    * @param fields - Fields to update
    * @param requests - Requests to update
-   * @param manager - An optional EntityManager which if provided *must* be used for all database
-   *   access. This is needed when creating anchors as part of a larger database transaction.
+   * @param options
+   * @returns A promise that resolves to the number of updated requests
    */
   public async updateRequests(
     fields: RequestUpdateFields,
     requests: Request[],
-    manager?: EntityManager
-  ): Promise<UpdateResult> {
-    if (!manager) {
-      manager = this.connection.manager
-    }
+    options: Options = {}
+  ): Promise<number> {
+    const { connection = this.connection } = options
+    const updatedAt = new Date(Date.now())
     const ids = requests.map((r) => r.id)
-    const result = await manager
-      .getRepository(Request)
-      .createQueryBuilder()
-      .update(Request)
-      .set(fields)
-      .whereInIds(ids)
-      .execute()
-      .then((result) => {
-        requests.map((request) => {
-          logEvent.db({
-            type: 'request',
-            ...request,
-            ...fields,
-            createdAt: request.createdAt.getTime(),
-            updatedAt: new Date(Date.now()),
-          })
-        })
-        return result
+    const result = await connection(TABLE_NAME)
+      .update({ ...fields, updatedAt: updatedAt })
```

Comment:
This `...fields` thing could backfire IMO at some point by trying to update a non-existent property. Maybe pass the fields explicitly? I know it might be considered an unnecessary duplication, but I see the risk here.

Feel free to act on that as you wish though @stephhuynh18

Reply:
Hmm, that's true. Missed that type guard. It feels safer still to make the arguments explicit, but I am ok with leaving the code as it is.